tests/functional/macros/test_macros.py

```diff
@@ -0,0 +1,337 @@
+{% endmacro %}
+"""
+
+macros__deprecated_adapter_macro = """
+{% macro some_macro(arg1, arg2) -%}
+    {{ adapter_macro('some_macro', arg1, arg2) }}
+{%- endmacro %}
+"""
+
+
+class TestMacros(object):
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self, project):
+        project.run_sql_file(project.test_data_dir / Path("seed.sql"))
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dep_macro.sql": models__dep_macro,
+            "local_macro.sql": models__local_macro,
+            "ref_macro.sql": models__ref_macro,
+        }
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"my_macros.sql": macros__my_macros}
+
+    @pytest.fixture(scope="class")
+    def packages(self):
+        return {
+            "packages": [
+                {
+                    "git": "https://github.com/dbt-labs/dbt-integration-project",
+                    "revision": "dbt/1.0.0",
+                },
+            ]
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "vars": {
+                "test": {
+                    "test": "DUMMY",
+                },
+            },
+            "macro-paths": ["macros"],
+        }
+
+    def test_working_macros(self, project):
+        run_dbt(["deps"])
+        results = run_dbt()
+        assert len(results) == 6
+
+        check_relations_equal(project.adapter, ["expected_dep_macro", "dep_macro"])
+        check_relations_equal(project.adapter, ["expected_local_macro", "local_macro"])
+
+
+class TestInvalidMacros(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dep_macro.sql": models__dep_macro,
+            "local_macro.sql": models__local_macro,
+            "ref_macro.sql": models__ref_macro,
+        }
+
+    def test_invalid_macro(self, project):
+        run_dbt(expect_pass=False)
+
+
+class TestAdapterMacroNoDestination(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__with_undefined_macro}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"my_macros.sql": macros__no_default_macros}
+
+    def test_invalid_macro(self, project):
+        with pytest.raises(dbt.exceptions.CompilationException) as exc:
+            run_dbt()
+
+        assert "In dispatch: No macro named 'dispatch_to_nowhere' found" in str(exc.value)
+
+
+class TestMacroOverrideBuiltin(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__override_get_columns_macros}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"macros.sql": macros__override_get_columns_macros}
+
+    def test_overrides(self, project):
+        # the first time, the model doesn't exist
+        run_dbt()
+        run_dbt()
+
+
+class TestMacroOverridePackage(object):
```

Comment:
This test involves an adapter repo but I'm sure it's an "adapter" test that should be factored out per se. Would this be best as a postgres test? Aren't those technically just core tests? I'm not seeing the reason for doing something special here.